lib/sda/treenode.go

```diff
@@ -0,0 +1,507 @@
+package sda
+
+import (
+	"errors"
+	"fmt"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/crypto/abstract"
+	"reflect"
+	"sync"
+)
+
+type TreeNodeInstance struct {
+	overlay *Overlay
+	token   *Token
+	// cache for the TreeNode this Node is representing
+	treeNode *TreeNode
+	// channels holds all channels available for the different message-types
+	channels map[network.MessageTypeID]interface{}
+	// registered handler-functions for that protocol
+	handlers map[network.MessageTypeID]interface{}
+	// flags for messages - only one channel/handler possible
+	messageTypeFlags map[network.MessageTypeID]uint32
```

Comment:
From my point of view, I don't think they're gonna be used in the from_scratch branch, since we have this notion of `Processor` interface which can be implemented by `ProcessorHandler` or `ProcessorChannels` or `ProcessorXXX`, the interface themselves will define how to dispatch the messages.
But in this PR I'd keep them as the tests are there to test this functionality and it's working so. Let's finish building this and then we can break other stuff :D